infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -265,171 +250,252 @@ public static void main(String[] args) {
+        } catch (InterruptedException ie) {
+            throw new UnrecoverableCorfuInterruptedError(ie);
+        }
+    }
+
+    /** Configure server bootstrap per-channel options, such as TCP options, etc.
+     *
+     * @param context       The {@link ServerContext} to use.
+     * @param bootstrap     The {@link ServerBootstrap} to be configured.
+     */
+    public static void configureBootstrapOptions(@Nonnull ServerContext context,
+            @Nonnull ServerBootstrap bootstrap) {
+        bootstrap.option(ChannelOption.SO_BACKLOG, 100)
+            .childOption(ChannelOption.SO_KEEPALIVE, true)
+            .childOption(ChannelOption.SO_REUSEADDR, true)
+            .childOption(ChannelOption.TCP_NODELAY, true)
+            .childOption(ChannelOption.ALLOCATOR, PooledByteBufAllocator.DEFAULT);
+    }
+
+    /** Get a "boss" group, which services (accepts) incoming connections.
+     *
+     * @param context       The {@link ServerContext} to use.
+     * @return              A boss group.
+     */
+    public static EventLoopGroup getBossGroup(@Nonnull ServerContext context) {
+        final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                                                    .setNameFormat("accept-%d")
+                                                    .build();
+        EventLoopGroup group = new NioEventLoopGroup(1, threadFactory);
+        log.info("getBossGroup: Type {}", group.getClass().getSimpleName());
+        return group;
+    }
+
+    /** Get a "worker" group, which services incoming requests.
+     *
+     * @param context   The {@link ServerContext} to use.
+     * @return          A worker group.
+     */
+    public static @Nonnull EventLoopGroup getWorkerGroup(@Nonnull ServerContext context) {
+        final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                .setNameFormat("worker-%d")
+                .build();
+
+        final int requestedThreads =
+                Integer.parseInt(context.getServerConfig(String.class, "--Threads"));
+        final int numThreads = requestedThreads == 0
+                                ? Runtime.getRuntime().availableProcessors() * 2
+                                    : requestedThreads;
+        EventLoopGroup group = new NioEventLoopGroup(numThreads, threadFactory);
+        log.info("getWorkerGroup: Type {} with {} threads",
+                        group.getClass().getSimpleName(), numThreads);
+        return group;
+    }
+
+
+    /** Obtain a {@link ChannelInitializer} which initializes the channel pipeline
+     *  for a new {@link ServerChannel}.
+     *
+     * @param context   The {@link ServerContext} to use.
+     * @param router    The {@link NettyServerRouter} to initialize the channel with.
+     * @return          A {@link ChannelInitializer} to intialize the channel.
+     */
+    private static ChannelInitializer getServerChannelInitializer(@Nonnull ServerContext context,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)